packages/components/dialog/src/DialogContent.styles.tsx

```diff
@@ -1,7 +1,12 @@
 import { cva, VariantProps } from 'class-variance-authority'
 
 export const dialogContentStyles = cva(
-  ['z-modal flex flex-col bg-surface group', 'focus-visible:outline-none focus-visible:u-ring'],
+  [
+    'z-modal flex flex-col bg-surface group',
+    'focus-visible:outline-none focus-visible:u-ring',
+    '[&:not(:has(footer))]:pb-lg',
+    '[&:not(:has(header))]:pt-lg',
```

Comment:
If I understand correctly, this would be broken if we use another tag element than `header|footer`? Why not simply using `first-child|last-child` instead? Might be more robust here, wdyt?

Reply:
Since `Dialog.Header` and `Dialog.Footer`serve as semantic header and footer elements respectively, it's unlikely that we would use div or other HTML elements instead

Reply:
Even with `asChild` API?

Reply:
If we need to detect that header/footer are not used in the future and the css selector is not enough, we could use the context